fuzz/fuzz_targets/cmos.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_main]
+use devices::legacy::Cmos;
+use libc::EFD_NONBLOCK;
+use libfuzzer_sys::fuzz_target;
+use vm_device::BusDevice;
+use vmm_sys_util::eventfd::EventFd;
+
+fuzz_target!(|bytes| {
+    // Need at least 16 bytes for the test
+    if bytes.len() < 16 {
+        return;
+    }
+
+    // Need multiples of 4 bytes to fill out the test
+    if bytes.len() % 4 != 0 {
+        return;
+    }
+
+    let mut below_4g = [0u8; 8];
+    let mut above_4g = [0u8; 8];
+
+    below_4g.copy_from_slice(&bytes[0..8]);
+    above_4g.copy_from_slice(&bytes[8..16]);
+
+    let mut cmos = Cmos::new(
+        u64::from_le_bytes(below_4g),
+        u64::from_le_bytes(above_4g),
+        EventFd::new(EFD_NONBLOCK).unwrap(),
+    );
+
+    let mut i = 0;
+    while i < bytes[16..].len() {
+        let offset = i;
+        let data = i + 1;
+        cmos.write(0, bytes[offset].into(), &bytes[data..data + 1]);
```

Comment:
Similarly, constraining the input `offset` to be valid (e.g. `bytes[offset].into() % 2`) would be better?